src/hooks/useRelatedVenues.tsx

```diff
@@ -59,19 +60,17 @@ export const RelatedVenuesProvider: React.FC<RelatedVenuesProviderProps> = ({
     error: relatedVenuesError,
     value: relatedVenues = emptyArray,
   } = useAsync(async () => {
-    if (!sovereignVenueId) return emptyArray;
+    if (!sovereignVenueId || !sovereignVenue) return emptyArray;
 
     return tracePromise(
       "RelatedVenuesProvider::fetchRelatedVenues",
-      () => fetchRelatedVenues(sovereignVenueId),
+      () => fetchRelatedVenues(sovereignVenue),
       {
-        attributes: {
-          sovereignVenueId: sovereignVenueId,
-        },
+        attributes: { sovereignVenueId },
         withDebugLog: true,
       }
     );
-  }, [sovereignVenueId]);
+  }, [sovereignVenueId, sovereignVenue]);
```

Comment:
remove sovereignVenueId from the deps and use only sovereignVenue please

Reply:
@mike-lvov i wanted to do it in that way but there appears an error. Do I need to correct the types tracePromise?
<img width="711" alt="Screenshot 2021-07-05 at 18 09 00" src="https://user-images.githubusercontent.com/86356678/124492334-fb060d00-ddbc-11eb-855c-dc37b861e794.png">

Reply:
could you not export sovereignVenueId and use the id inside since it's anyway used in the fn

Reply:
Why do we want to change this? Fetch related venues should only receive the id, the attributes should only have the id, this particular snippet shouldn’t have the full venue in it at all..

Reply:
The solution to that typing bug is to read the id off the sovereignVenue

Reply:
> Why do we want to change this? Fetch related venues should only receive the id, the attributes should only have the id, this particular snippet shouldn’t have the full venue in it at all..

My intention was not to double fetch the sovereignVenue since we already have it in place. It would improve the performance especially for the lower-level venues

Reply:
This change makes what is currently an all in one wrapped up good generic function, dependendent on having an entire venue object available to it and passed as a prop.. that's not a good way to do things.. to say the least.

Reply:
@0xdevalias @mike-lvov fixed in 98b0388a5f7e0fb23799f214e5b588b02eaf1fa0. If I understood correctly, I needed to set sovereignVenueId instead of sovereignVenue.id  in tracePromise function, I did so.